File: API/Domain/Asset/Asset.cs
using API.Domain.Shared;

namespace API.Domain.Asset;

public class Asset : TenantEntity
{
    public required string Description { get; set; }
    public List<AssetTag> Tags { get; set; } = [];

    public required string CategoryId { get; set; }
    public AssetCategory? Category { get; set; }

    public int StockLevel { get; set; }

    public List<AssetSupplySource> SupplySources { get; set; } = [];
    public List<AssetStockLevel> HistoricStockLevels { get; set; } = [];
}

File: client/src/core/data/entities/asset/asset.ts
import { AssetCategory } from './asset-category.ts';
import { BaseEntity } from '../base-entity.ts';
import { AssetSupplySource } from './asset-supply-source.ts';
import { AssetStockLevel } from './asset-stock-level.ts';

export interface Asset extends BaseEntity {
  category?: AssetCategory;
  description: string;
  tags: AssetTag[];
  stockLevel: number;
  supplySources: AssetSupplySource[];
  historicStockLevels: AssetStockLevel[];
}

export interface AssetTag {
  id: string;
  tag: string;
}

export interface AssetForm {
  id?: string;
  categoryId: string;
  description: string;
  stockLevel: number;
  tags?: string[];
}